### PR TITLE
dimensionned units for multiply/division 

### LINKED ISSUE #109: Add unit support in multiply and other spectrum operations
### Problem

As mentioned in https://github.com/radis/radis/pull/101#pullrequestreview-394154396, the problem with the normalization of #101  is that [multiply](https://github.com/radis/radis/blob/develop/radis/spectrum/operations.py#L480) does not use dimensioned units. 

> multiply does not currently handle multiplications with dimensioned coefficients. These should change the units of the output Spectrum.

### Suggested fix

(this is related to #57 , which focused on Spectrum methods, while here we have operations between 2 Spectra, or 1 Spectrum or arrays/floats) 

- [x] use astropy.units for the conversion. We now have (#106 ) a radis.phys.units.Unit based on astropy.units:
```
a = b * c 
unit_a = (Unit(unit_b) * Unit(unit_c)).to_string() 
```

- [x] add tests!
Such as: 
```
Radiance(s) -= 0.01 * Unit("W/cm2/sr/µm")
```

- [x] same for all the others: add, divide, etc.

Finally, some cleaning:

- [x] Implement a new normalize() method with the different options of get_residual() (that uses our new dimensioned multiply method)
- [ ] Replace most of the normalizing code of get_residual() with a simple call to `get_diff(s1.normalize(*args), s2.normalize(*args) )` ? 
- [ ] that should eventually fix #102. Check and close Issue.

- [ ] Also i think we can do the same in .plot() , if we know have a normalize method()

Final remark:

how do you normalize a transmittance spectrum? It does not really make sense, but is not forbidden here. Should raise a NotImplementedError or ValueError.

### PR DESCRIPTION
**Pull request by erwanp**
### Description

- Add unit support for multiply / division. Fixes #109 
- Also add min(), max() methods for spectra with unique quantities 

Allows to use the following syntax : 
```
# add a baseline : 
s += 0.1 * u.Unit("W/cm2/sr/nm")   

# normalize manually : 
s /= s.max() * u.Unit("mW/cm2/sr/nm")   
assert s.units['radiance'] == "1.0" # normalized
assert s.max() == 1.0

# Calibrate :
s.units['radiance'] = 'count'
s *= 100 * u.Unit("mW/cm2/sr/nm/count")

```

Edit : 

also added a normalization method directly : 

```

    s.normalize()
    s.normalize(normalize_how="max")
    s.normalize(normalize_how="area")
    
You can also normalize only on limited range. Useful for noisy spectra ::

    s.normalize(wrange=(2250, 2500), wunit="cm-1", normalize_how="mean")    

This returns a new spectrum and does not modify the Spectrum itself. To do so use::

    s.normalize(inplace=True)
    
You can chain the commands :: 

    s.normalize().plot()
```

### CODE REVIEW

File: radis/spectrum/operations.py [hunk 1]
```diff
@@ -477,15 +463,18 @@ def _get_unique_var(s, var, inplace):
...
 
     Parameters
     ----------
-    s: Spectrum objects
-        The spectra to multiply.
+    s: Spectrum object
+        The spectrum to multiply.
     coef: float
         Coefficient of the multiplication.
+    unit: str.
```
Comment (erwanp) [inline radis/spectrum/operations.py:475]:
```suggestion
    unit: str, or `~astropy.units.core.Unit`
```

Comment (erwanp) [reply radis/spectrum/operations.py:475]: (misc) improve docs

File: radis/spectrum/spectrum.py [hunk 2]
```diff
@@ -3574,6 +3575,159 @@ def __str__(self):
...
+                norm = np.abs(nantrapz(I[b], w[b]))
+                norm_unit = u.Unit(s.units[var]) * u.Unit(wunit)
+            else:
+                raise ValueError(
+                    "Unexpected `normalize_how`: {0}".format(normalize_how)
+                )
+
+            out = multiply(s, 1 / norm, unit=norm_unit, inplace=inplace)
+
+        else:
```
Comment (minouHub) [inline radis/spectrum/spectrum.py:3707]: (details)
I would make clearer the if ... then ... else here


```
 if len(wrange) == 2:
             blabla
 elif isinstance(wrange, (int, float)):
             blabla
  else:
             UserWarning

```

**Review by minouHub — APPROVED** [review]
Except the minor comment about the `if .. else` everything is ok for me

File: radis/spectrum/spectrum.py (hunk 2 above)
Comment (erwanp) [reply radis/spectrum/spectrum.py:3707]: Great, and it would also to normalize at one specific point. I forgot this one :)

**Comment by erwanp** [comment]
> 
> 
> Except the minor comment about the `if .. else` everything is ok for me

no time to implement it now, i opened a new Issue #127 and I'm merging this one now !

**Pull request merged**